rtCamp/next-pms
# refactor: Create generic component for handling the custom views

> Please provide enough information so that others can review your pull request:
- This pr focuses on creating generic component for handling the custom view to reduce the duplicate code, also for all the list view we need to have common component for showing the data in table cell based on the meta data of the field.

> Explain the **details** for making this change. What existing problem does the pull request solve?
- Create `ViewWrapper` component for custom view handling.
- Create `DataCell` component to render the cell based on the meta information. (Every listview can have their specific implementation for rendering the cell if there is any specific thing required.)
- Do not rigger zod schema validation while adding new row in `EditTime` dialog. 
- Remove extra width from user avatar in sidebar.
- Organize import and rename jsx component file.

> QA instruction
1. Login as employee.
2. Go to `Task` or `Project` page
3. Create custom view for each page and check if the filters are being applied correctly.

> QA List
- [ ] Check if custom views are working properly on `Task` and `Project` page.

> Screenshots/GIFs
- Not applicable since functionality is already built.

ref: 
- https://github.com/rtCamp/next-pms/issues/305
